src/main/java/com/bobocode/hoverla/bring/context/BeanDefinitionValidator.java

```diff
@@ -148,6 +149,10 @@ private BeanDefinition resolveDependency(String dependencyName, Class<?> depende
             foundDependency = definitionByName.get();
             checkTypeMatching(dependencyType, foundDependency);
         } else {
+            if (!StringUtils.equals(dependencyName, dependencyType.getName())) {
```

Comment:
Please, leave a comment in code explaining why do we need this, for better readability

Reply:
done

Reply:
@lastvova why do we need `StringUtils` here?

Reply:
I thought it will be a good approach. If String.equals() is enough, I will change it.